src/js/game/core.js

```diff
@@ -1,573 +1,583 @@
+
+        // Initialize canvas element & context
+        this.internalInitCanvas();
+
+        // Members
+        const root = this.root;
+
+        // This isn't nice, but we need it right here
+        root.keyMapper = new KeyActionMapper(root, this.root.gameState.inputReciever);
+
+        // Init game mode
+        root.gameMode = GameMode.create(root, gameModeId, parentState.creationPayload.gameModeParameters);
+
+        // Needs to come first
+        root.dynamicTickrate = new DynamicTickrate(root);
+
+        // Init classes
+        root.camera = new Camera(root);
+        root.map = new MapView(root);
+        root.logic = new GameLogic(root);
+        root.hud = new GameHUD(root);
+        root.time = new GameTime(root);
+        root.achievementProxy = new AchievementProxy(root);
+        root.automaticSave = new AutomaticSave(root);
+        root.soundProxy = new SoundProxy(root);
+
+        // Init managers
+        root.entityMgr = new EntityManager(root);
+        root.systemMgr = new GameSystemManager(root);
+        root.shapeDefinitionMgr = new ShapeDefinitionManager(root);
+        root.hubGoals = new HubGoals(root);
+        root.productionAnalytics = new ProductionAnalytics(root);
+        root.buffers = new BufferMaintainer(root);
+
+        // Initialize the hud once everything is loaded
+        this.root.hud.initialize();
+
+        // Initial resize event, it might be possible that the screen
+        // resized later during init tho, which is why will emit it later
+        // again anyways
+        this.resize(this.app.screenWidth, this.app.screenHeight);
+
+        if (G_IS_DEV) {
+            // @ts-ignore
+            window.globalRoot = root;
+        }
+
+        // @todo Find better place
+        if (G_IS_DEV && globalConfig.debug.manualTickOnly) {
+            this.root.gameState.inputReciever.keydown.add(key => {
+                if (key.keyCode === 84) {
+                    // 'T'
+
+                    // Extract current real time
+                    this.root.time.updateRealtimeNow();
+
+                    // Perform logic ticks
+                    this.root.time.performTicks(this.root.dynamicTickrate.deltaMs, this.boundInternalTick);
+
+                    // Update analytics
+                    root.productionAnalytics.update();
+
+                    // Check achievements
+                    root.achievementProxy.update();
+                }
+            });
+        }
+
+        logger.log("root initialized");
+    }
+
+    /**
+     * Initializes a new game, this means creating a new map and centering on the
+     * playerbase
+     *
+     * @param {object} param1
+     * @param {number} param1.seed
+     * */
+    initNewGame({ seed }) {
+        logger.log("Initializing new game");
+        this.root.gameIsFresh = true;
+
+        if (seed === undefined) {
+            this.root.map.seed = randomInt(0, 100000);
+        } else {
+            this.root.map.seed = seed;
+        }
```

Comment:
```suggestion
    initNewGame({ seed = randomInt(0, 100000) }) {
        logger.log("Initializing new game");
        this.root.gameIsFresh = true;

        this.root.map.seed = seed;
```
that should be how a default parameter is done, but note I did not test this.

Reply:
Ill test it right now!

Reply:
what exactly does that mean?